R/plumber.R

```diff
@@ -4,69 +4,80 @@ NULL
 
 # used to identify annotation flags.
 verbs <- c("GET", "PUT", "POST", "DELETE", "HEAD", "OPTIONS", "PATCH")
-enumerateVerbs <- function(v){
-  if (identical(v, "use")){
+enumerateVerbs <- function(v) {
+  if (identical(v, "use")) {
     return(verbs)
   }
   toupper(v)
 }
 
 #' @rdname plumber
 #' @export
-plumb <- function(file, dir="."){
-
-  dirMode <- NULL
-
-  if (!missing(file) && !missing(dir)){
-    # Both were explicitly set. Error
-    stop("You must set either the file or the directory parameter, not both")
+plumb <- function(file = NULL, dir = ".") {
+
+  if (!is.null(file) && !identical(dir, ".")) {
+    # both were explicitly set.
+    # assume it is a file in that dir and continue like normal
+    file <- file.path(
+      # removing trailing slash in dir
+      sub("/$", "", dir),
+      file
+    )
+  }
 
-  } else if (missing(file)){
-    if (identical(dir, "")){
+  if (is.null(file)) {
+    if (identical(dir, "")) {
       # dir and file are both empty. Error
       stop("You must specify either a file or directory parameter")
     }
```

Comment:
Seems like this would be rendered unnecessary with a `dir <- normalizePat(dir, mustWork = TRUE)`

Reply:
Going to keep the error as is.  The message is more useful than seeing `path:'' can not be found`.